app/models/spree/gateway/stripe_sca.rb

```diff
@@ -79,6 +79,11 @@ def create_profile(payment)
 
       private
 
+      def refundable_amount(payment_intent_response)
+        payment_intent_response.amount_received -
+          payment_intent_response.charges.data.map(&:amount_refunded).sum
```

Comment:
What if there are unsuccessful charges? Looking at the docs it seems that they might not even include them all?

> This list only contains the latest charge, even if there were previously multiple unsuccessful charges. To view all previous charges for a PaymentIntent, you can filter the charges list using the payment_intent parameter.

source: https://stripe.com/docs/api/payment_intents/object#payment_intent_object-charges.

Reply:
I believe this still works, because each time we try to make a payment for an order it creates a new PaymentIntent. I tried a couple things:

At first, I thought you meant "what if there are multiple partial refunds" so I tried that: I placed an order, then in the backend I made two adjustments to the order and issued a refund each time. The `amount_refunded` returns the sum across all refunds. 

Then I realized you might be asking "what if the customer or an admin creates unsuccessful charges?" For the admin side, I placed an order through the front end and chose Cash. Then on the backend, I created two unsuccessful charges and one successful one. I was still able to do the partial refunds on the successful charge. I found the same when the unsuccessful charges were created by the customer on the front end. 

The option to do the refund on the failed charge never appears in the first place, only on the successful one: 
<img width="707" alt="Screen Shot 2020-12-10 at 2 37 57 PM" src="https://user-images.githubusercontent.com/35588/101838356-4db46a80-3af5-11eb-891c-03ce68797cdf.png">

Reply:
But does that mean that the `payment_intent_response.charges` includes only successful charges then? Sorry, I'm still wrapping my head around this Stripe API.

Reply:
What I'm saying is that for our case, the PaymentIntent that we're querying Stripe about will only have the single successful charge. It's definitely complex! Especially since our mapping is not as direct as it could be if we were only dealing with Stripe as a provider.